iterator.md

```diff
@@ -0,0 +1,67 @@
+# Iterator Pattern
+
+Iteratoren werden in der C# Entwicklung als Liste oder Arrays verwendet und beispielsweise als Interface `IEnumerable`
```

Comment:
Ich hab hier das nochmal durchgelesen. Hier geht es doch um das Iterieren über eine Liste. Aber das Iterator Entwurfsmuster macht doch was anderes? https://de.wikipedia.org/wiki/Iterator_(Entwurfsmuster)

Reply:
mhh.. ok so i have explained the wrong function or the wrong iterator? I already found this on the official Microsoft webpage. So i must explain the Iterator, which you have provided in you link correct?

Reply:
ja